darts/timeseries.py

```diff
@@ -346,6 +394,14 @@ def from_csv(
             fill in missing values if some dates are missing and `fill_missing_dates` is set to `True`.
         fillna_value
             Optionally, a numeric value to fill missing values (NaNs) with.
+        static_covariates
+            Optionally, a set of static covariates to be added to the TimeSeries. Either a pandas Series or a pandas
+            DataFrame. If a Series, the index represents the static variables. The covariates are globally 'applied'
+            to all components of the TimeSeries. If a DataFrame, the columns represent the static variables and the
+            rows represent the components of the uni/multivariate TimeSeries. If a single-row DataFrame, the covariates
+            are globally 'applied' to all components of the TimeSeries. If a multi-row DataFrame, the number of
```

Comment:
When you write "If a single-row DataFrame, the covariates are globally applied" --> That's a good idea. But is it implemented? In the constructor, I only see a check that the length of the DataFrame matches the number of components. Maybe we should "tile" the DataFrame if it has length 1?

Reply:
In the constructor it checks the following:
https://github.com/unit8co/darts/blob/6dc7ff886099e775699e5e2dd60fbf8ccd38e75e/darts/timeseries.py#L220

so either the df has 1 or `n_components` components.

The component names are then set in the following lines:
https://github.com/unit8co/darts/blob/6dc7ff886099e775699e5e2dd60fbf8ccd38e75e/darts/timeseries.py#L235

When the number of static cov components is equal to `n_components`, it simply assign the component names to the static covs.
If the number is not equal `n_components`, it must be a single global component and it gets the following component name:
https://github.com/unit8co/darts/blob/6dc7ff886099e775699e5e2dd60fbf8ccd38e75e/darts/timeseries.py#L55

I don't think we need to tile the DataFrame as it doesn't add any new information. WDYT?

If we have a multivariate series with the global staic covariates and we select on component from the series, the static covariate component name will change from `DEFAULT_GLOBAL_STATIC_COV_NAME ` to the name of the univariate component:

```
>>> sc = pd.Series([0., 1.], index=["sc1", "sc2"])
>>> ts = TimeSeries.from_values(values=np.random.rand(5, 2), columns=["a", "b"]).with_static_covariates(sc)
>>> print(ts.static_covariates)
>>> print("")
>>> print(ts["a"].static_covariates)

static_covariates  sc1  sc2
global_components  0.0  1.0

static_covariates  sc1  sc2
component                  
a                  0.0  1.0
```

Reply:
OK I see. I had in mind to tile it so that whenever we access the covariates values (e.g. from training datasets), we always get a tensor with dimension matching the number of components.